### ISSUE TITLE
Suggestion: Include motivation in README.md.

### ISSUE THREAD

**Issue by Julian-O** [globality-corp/flake8-logging-format#15]
The README.md currently says:

>  One way to ensure consistency and rigor in logging is to always use extra to pass non-constant data and, therefore, to never use format strings, concatenation, or other similar techniques to construct a log string.

However, if doesn't explain *why* this rigor is important. It is easy to dismiss this: "A foolish consistency is the hobgoblin of little minds."

I would add a few paragraphs explaining *why* this practice should be adopted.

e.g.

> It is important to avoid using format strings for two reasons:
>    1. Performance: If a log message is turned off, it is unnecessary to perform the string interpolation and/or the string concatenation and the implicit calls to `__str__()` functions on the parameters. Keeping the parameter separate means that the logging system can skip such operations.
>   2. Maintainability: The `logging` module is designed to provide a separation of concerns: the treatment of log messages from a module can be modified outside of the module.  In particular, `logging.Handler` and `logging.Filter` classes can be customised to identify particular log messages and take particular actions, such as filter them, escalate them, replace them with a clearer log message or translate them to a different language. These manipulations are easier if access is given to the raw log message *before* the string interpolation is performed.

**Comment by fpuga** [comment]
It seems that the original motivation was explained in #5 but i agree that it will be nice have it in the README.

Also, i don't understand the example in the README. Afaik `extra` keyword is not used for [variable data](https://docs.python.org/3/howto/logging.html#logging-variable-data) but for the `logging.Formatter`.

```python
>>> import logging

>>> logging.error("Hello {world}", extra={"world":"Earth", "color": "blue"})                                                                                                                                                   
ERROR:root:Hello {world}

>>> import logging 
>>> logging.basicConfig(format="%(color)s - %(message)s") 
>>> logging.error("Hello {world}", extra={"world":"Earth", "color": "blue"})                                                                                                                                   
blue - Hello {world}
```

**Comment by afallou** [comment]
@fpuga that is correct - in our case we use two custom formatters. [One we wrote](https://github.com/globality-corp/microcosm-logging/blob/develop/microcosm_logging/formatters.py#L6) for console output, that will `format` the log message using those extra arguments. [The other from `pythonjsonlogger`](https://github.com/madzak/python-json-logger/blob/master/src/pythonjsonlogger/jsonlogger.py#L77) will add the `extra` dict to the logged JSON.
I agree the README could be made clearer there.

**Comment by afallou** [comment]
@Julian-O a paragraph on motivation was added to the `README` in https://github.com/globality-corp/flake8-logging-format/pull/19

**Comment by fpuga** [comment]
Thanks @afallou, it's more clear now.

**Comment by FanchenBao** [comment]
Since it is clear now that `extra` dict is only used for formatter, please also update the example. It could cause confusion.

**Comment by Dreamsorcerer** [comment]
Has caused confusion. Just spent half an hour trying to figure out why it the variables weren't being output.
